private/buf/bufgen/generator.go

```diff
@@ -209,34 +208,12 @@ func (g *generator) execPlugins(
 	for remote, indexedPluginConfigs := range remotePluginConfigTable {
 		remote := remote
 		indexedPluginConfigs := indexedPluginConfigs
-		v1Args := make([]*remotePluginExecArgs, 0, len(indexedPluginConfigs))
 		v2Args := make([]*remotePluginExecArgs, 0, len(indexedPluginConfigs))
 		for _, param := range indexedPluginConfigs {
-			if param.PluginConfig.Plugin == "" {
```

Comment:
Don't we still need to support this in the config file? Or is that logic in the external -> internal config stuff?

Reply:
The `Plugin` key is still there, but I flipped the logic to check against the `remote` key to then use `param.PluginConfig.Remote` in the error message.

We could also push this much earlier to the external -> internal validation and remove `Remote` from internal packages entirely?

https://github.com/bufbuild/buf/blob/6c98e0e88168c6e17b8c82f489ab68cd705b943e/private/buf/bufgen/config.go#L211-L217

Reply:
Added a check in https://github.com/bufbuild/buf/pull/2278/commits/0ec5f4aa325c715f0644675617f0a0c914428a3b, I think there's an opportunity to cleanup `bufgen/{config,generator}.go`, but I suggest we do this in a separate PR or fold it into the v2 work.

We don't need the internal Remote config, and instead gate it at the external config layer.

### Example 1

```yaml
plugins:
  - remote: buf.build/protocolbuffers/plugins/go
  # or
  - remote: buf.build/protocolbuffers/go
```

We surface an error unconditionally for both new and old references since the `remote` field is being used in the external config.

> Failure: File "buf.gen.yaml": the remote field is no longer supported, see the migration guide: https://buf.build/docs/migration-guides/migrate-remote-generation-alpha/#migrate-to-remote-plugins

### Example 2

```yaml
plugins:
  - plugin: buf.build/protocolbuffers/plugins/go
```

> Failure: File "buf.gen.yaml": invalid plugin reference: buf.build/protocolbuffers/plugins/go